### PR TITLE
Dataset: Rename Dataset class to DatasetPackage and update properties desc.

### LINKED ISSUE #657: Dataset: Is it a Dataset class or a DatasetPackage class?
- In Profile conformance information https://github.com/spdx/spdx-3-model/blob/main/model/Dataset/Dataset.md , it mentioned `/Dataset/DatasetPackage`
- In https://github.com/spdx/spdx-3-model/blob/main/model/Dataset/Classes/Dataset.md , the class that is SubclassOf: `/Software/Package` is named `Dataset`

(For AI Profile, its main (and only) class is `AIPackage`.)

### PR DESCRIPTION
**Pull request by bact**
**Rename `Dataset` class to `DatasetPackage`:**
- To fix #657 to maintain consistency (consensus reached in AI Team Meeting 6 March 2024)
- Update the label in the diagram `model.drawio` accordingly
- Update descriptions of `Dataset.md`, `DatasetPackage.md`, `AI.md`, `AIPackage.md` to reflect the new `DatasetPackage` name.
  - Also differentiate between the Profile and the xxPackage class.

**Update other property descriptions:**
- Remove the property names from the beginning of the first paragraph in the description.
  - This is to avoid confusion about actual casing
  - For example, `autonomyType` (actual property name) vs "AutonomyType" (the mention of the property in the beginning of the paragraph).
    - The first one is camelCase (small A at the beginning), the latter is PascalCase (capital A).
  - There was at least one inconsistency as well. `datasetType` was mentioned as "Type". Removing the mention also fixed the inconsistency.
- For properties with `Range: xsd:string`, add "A free form text that ..." to the description.
- Remove trailing spaces at the end of lines and newlines at the end of files.

### CODE REVIEW

**Comment by bact** [comment]
@kestewart please kindly review.

Some changes (description editing) are opportunistic, I can remove them to make the change set more minimal.

fyi @rgopikrishnan91 @bennetkl I have moved the PR here after trying resolve merge conflicts.

File: model/AI/AI.md [hunk 1]
```diff
@@ -4,18 +4,18 @@ SPDX-License-Identifier: Community-Spec-1.0
 
 ## Summary
 
-Additional metadata based on software profile, that is useful for ai applications and models.
+The AI Profile provides additional metadata, based on Software Profile, that is useful for AI applications and models.
```
Comment (bennetkl) [inline model/AI/AI.md:7]: @kestewart  @bact  Could we change summary AI profile definition to The AI profile is designed to provide a standardized way of documenting and sharing information about AI software packages (ie. system)

File: model/AI/AI.md [hunk 2]
```diff
@@ -4,18 +4,18 @@ SPDX-License-Identifier: Community-Spec-1.0
...
 
 ## Description
 
-The AI profile namespace defines concepts related to AI application and model artifacts.
+The AI namespace defines concepts related to AI application and model artifacts.
```
Comment (bennetkl) [inline model/AI/AI.md:11]: Could we change to he AI namespace defines a set of concepts and data elements related to AI system  and model artifacts. These artifacts are the tangible outputs of the AI development process, such as software packages, models, and data sets.

File: model/AI/Properties/energyConsumption.md [hunk 3]
```diff
@@ -8,7 +8,7 @@ Indicates the amount of energy consumed to build the AI package.
 
 ## Description
 
-EnergyConsumption captures the amount of energy needed to train and operate the AI model. 
+A free form text that captures the amount of energy needed to train and operate the AI model.
 This value is also known as training energy consumption or inference energy consumption.
```
Comment (bennetkl) [inline model/AI/Properties/energyConsumption.md:12]: We should add definition of training and inference, ie. Training Energy Consumption is the amount of energy consumed during the training phase of an AI model. This includes the energy used to power the hardware and software systems used for training, such as GPUs, CPUs, and data center is
Inference Energy Consumption:is the amount of energy consumed during the inference phase of an AI model. This includes the energy used to power the hardware and software systems used for making predictions or decisions based on the model, such as edge devices, servers, or cloud services. Total Energy Consumption: is he total amount of energy consumed by an AI model over its entire lifecycle, including training, inference, and any other related activities

File: model/AI/Properties/hyperparameter.md [hunk 4]
```diff
@@ -8,7 +8,7 @@ Records a hyperparameter used to build the AI model contained in the AI package.
 
 ## Description
 
-This field records a hyperparameter value.
+Records a hyperparameter value.
 Hyperparameters are parameters of the machine learning model that are used to control the learning process,
 for example the optimization and learning rate used during the training of the model.
```
Comment (bennetkl) [inline model/AI/Properties/hyperparameter.md:13]: Suggest adding, a hyperparameter is a parameter that is set before the training process begins and is used to control the behavior of the learning algorithm. Hyperparameters are not learned from the data, but are instead set by the developer or chosen through a process of trial and error.

 add in the for example, learning rate, batch size, or number of layers.

File: model/AI/Properties/informationAboutApplication.md [hunk 5]
```diff
@@ -8,12 +8,11 @@ Provides relevant information about the AI software, not including the model des
 
 ## Description
 
-InformationAboutApplication describes any relevant information in free form text about 
-how the AI model is used inside the software, as well as any relevant pre-processing steps, third party APIs etc. 
+A free form text that describes any relevant information about
+how the AI model is used inside the software, as well as any relevant pre-processing steps, third party APIs etc.
```
Comment (bennetkl) [inline model/AI/Properties/informationAboutApplication.md:13]: We could add abit more ..  fro examples are :   functionality provided by the AI model within the software application, including any specific tasks or decisions it is designed to perform,  any pre-processing steps that are applied to the input data before it is fed into the AI model, such as data cleaning, normalization, or feature extraction, and  any third-party APIs or services that are used in conjunction with the AI model, such as data sources, cloud services, or other AI models.
    Dependencies: A description of any dependencies or requirements needed to run the AI model within the software application, such as specific hardware, software libraries, or operating systems.

File: model/AI/Properties/informationAboutTraining.md [hunk 6]
```diff
@@ -8,8 +8,8 @@ Describes relevant information about different steps of the training process.
 
 ## Description
 
-InformationAboutTraining describes the specific steps involved in the training of the AI model.
-For example, it can be specified whether supervised fine-tuning 
+A free form text that describes the specific steps involved in the training of the AI model.
+For example, it can be specified whether supervised fine-tuning
 or active learning is used as part of training the model.
```
Comment (bennetkl) [inline model/AI/Properties/informationAboutTraining.md:14]: maybe change to: a detailed explanation of the training process, including any specific techniques, algorithms, or methods used to train the model.  For example,   training data used to train the AI model, including any relevant details about the data source, data quality, and data pre-processing steps, specific training algorithm used to train the AI model, such as stochastic gradient descent, backpropagation, or reinforcement learning, any specific training techniques used to improve the performance or accuracy of the AI model, such as transfer learning, fine-tuning, or active learning and any evaluation metrics used to assess the performance of the AI model during the training process, such as accuracy, precision, recall, or F1 score.

Comment (bact) [reply model/AI/Properties/informationAboutTraining.md:19]: Done

File: model/AI/AI.md (hunk 2 above)
Comment (bact) [reply model/AI/AI.md:11]: Done

File: model/AI/AI.md (hunk 1 above)
Comment (bact) [reply model/AI/AI.md]: I have updated the PR as proposed

File: model/AI/Properties/hyperparameter.md (hunk 4 above)
Comment (bact) [reply model/AI/Properties/hyperparameter.md]: Done

File: model/AI/Properties/energyConsumption.md (hunk 3 above)
Comment (bennetkl) [reply model/AI/Properties/energyConsumption.md:13]: Yes, your assumption is correct. This field (energyconsumption) of training data was added since IBM has a tool that calculates that metric.  (IBM Japan research demo'es this tool at Linux Summit 2022).  We can create a separate PR to discuss how to track/calculate total energy consumption, something that we need to think about is keeping AI BOM dynamic, AI systems are not statis.  A bigger discussion  But I approve your proposed description changes. .

Comment (bact) [reply model/AI/Properties/energyConsumption.md]: @bennetkl I have created #676 for the discussion of `inferenceEnergyConsumption` and `totalEnergyConsumption`

**Review by kestewart — APPROVED** [review]
Thanks for the cleanup.

**Comment by rgopikrishnan91** [comment]
Looks good to me. I think can be merged. Thanks @bact !

**Comment by kestewart** [comment]
Gopi's ok is in comments, so going ahead and merging.
